src/main/java/dev/morling/onebrc/CalculateAverage_JesseVanRooy.java

```diff
@@ -0,0 +1,256 @@
+/*
+ *  Copyright 2023 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.onebrc;
+
+import sun.misc.Unsafe;
+
+import java.io.IOException;
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.ValueLayout;
+import java.lang.reflect.Field;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+import java.util.stream.IntStream;
+
+//Disclaimer: The idea from the segmentation into #core amount of chunks came from previously submitted solutions.
+public class CalculateAverage_JesseVanRooy {
+
+    private static final String FILE = "./measurements.txt";
+
+    private static final ValueLayout.OfByte DATA_LAYOUT = ValueLayout.JAVA_BYTE;
+
+    private static final Unsafe UNSAFE = initUnsafe();
+
+    private static Unsafe initUnsafe() {
+        try {
+            Field theUnsafe = Unsafe.class.getDeclaredField("theUnsafe");
+            theUnsafe.setAccessible(true);
+            return (Unsafe) theUnsafe.get(Unsafe.class);
+        }
+        catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static class Result {
+        long nameStart;
+        long nameSize;
+        String name;
+        int min;
+        int max;
+        long sum;
+        int count;
+
+        double min() {
+            return min / 10.0;
+        }
+
+        double max() {
+            return max / 10.0;
+        }
+
+        double mean() {
+            return (sum / 10.0) / count;
+        }
+    }
+
+    public static class ThreadResult {
+        Result[] results;
+    }
+
+    static final int MAP_SIZE = 4096;
+    static final int MAP_MASK = MAP_SIZE - 1;
+    static final int VALUE_CAPACITY = 512;
+
+    static void process(MemorySegment memorySegment, ThreadResult threadResult) {
+        // initialize hash table
+        final int[] keys = new int[MAP_SIZE];
```

Comment:
Is this resized somewhere? There can be up to 10K station names.

Reply:
No, it just wasn't properly sized for the requirements.
Fixed it.
It doesn't seem to affect performance on my machine, so it should still be try.

Reply:
I wonder though how this passed the 10K test case there is. Could you revert it to the previous value and see why that's the case? Maybe a bug still somewhere?

Reply:
It didn't, I forgot to run that test ... embarrasing.

I ran some of the tests manually intelliJ because I had trouble with UTF-8 encoding of the output when diffing it with the expected result. (I'm running this on Windows).

I now fixed the UTF-8 issues by adding '-Dsun.stdout.encoding=UTF-8' to the calculate_average_<forkName>.sh file.
This also revealed that adding %n does not print the expected \n in the output (it prints \r\n instead of \n), so I changed that too.

In order to verify that my output matches the output of the baseline program I had to do similar changes to the baseline program & script, but did not commit those as we are probably not intended to change these files.

I do hope the -Dsun.stdout.encoding=UTF-8 does not mess with the output on other OSes, but I would not expect it to.

If this doesn't affect the outcome on other OSes at all, you could consider adding this to the run script of the baseline and changing the println in the baseline script to a print, while manually adding the \n. This would improve compatibility on Windows.

Reply:
That's the thing, it passed that test for me, also with that old map size. Just tried it again by changing the size back to a value smaller than 10K. Can you reproduce this? Something seems odd here.

Reply:
Oh, you probably only changed the map size.
In that case it would work because there are still multiple thread using that map, and since the 10k test has exactly 10k lines, it would never give more than 4096 lines to one thread, hence not exceeding the map size.
If you also change the VALUE_CAPACITY to the previous value, it would fail.

Reply:
I actually thought about this some more:

Is it the case that the original solution (MAP_SIZE = 4096 and VALUE_CAPACITY = 512) worked for you?

Because in that case it still should have flagged a test failure on an 8 core machine (from the Readme: "Programs are run from a RAM disk (i.o. the IO overhead for loading the file from disk is not relevant), using 8 cores of the machine.").

Could it be the case that the test is being run with more cores then? Because the only way I see that the VALUE_CAPACITY = 512 solution works is if there are 20+ threads at work. And that would only happen if 'Runtime.getRuntime().availableProcessors()' returns more than 20.

Reply:
Ah yes indeed. Tests run with 32 cores, evaluation with 8 then, as per the rules.